src/MainWindow.cc

```diff
@@ -133,7 +133,7 @@ QWidget * MainWindow::findPage( const QString & pageName )
         return 0;
     }
 
-    foreach ( QWidgetPointer page, _pages )
+    for ( const QWidgetPointer& page : std::as_const( _pages ) )
```

Comment:
Let's not get overboard here. `QWidgetPointer` is typedef`ed to a `QPointer<QWidget>`. It's still a pointer type. It might carry a reference count in addition to the pointer itself, but it's still a very trivial data type, not anything that costs enormous memory overhead. Those types should not be used as a const reference; that only makes the code less readable.

And again, `std::const()` adds to making the code harder to read.

Reply:
Without the reference, it will create a copy of a `QWidgetPointer` in every loop iteration.